### PR TITLE
Minor improvements to release day docs

### PR DESCRIPTION
**Pull request by bhearsum**
* Fixed a link
* Moved things that happen on release day to their own section (current docs make it seem like they should happen at the same time as central/esr bumps)
* Simplified Ship It bump docs. IMO these are too wordy for what is supposed to be a how to. Feel free to disagree and push back though.
* Unrelated to merge day - removed a reference to a dead how-to.
* Added a note about trying to Ship It bumps on merge day part 2.

### CODE REVIEW

**Review by gabrielBusta — APPROVED** [review]

File: procedures/release-duty/merge-duty/merge_duty.rst [hunk 1]
```diff
@@ -379,23 +379,20 @@ Update the releng changelog
...
-The nightly builds trigger on the schedule defined in `.cron.yml <https://hg.mozilla.org/mozilla-central/file/8e5247451c9a83abe6f877f5f2c62332a84c9aaf/.cron.yml#l34>`__ (in UTC) and take a few hours to complete.
-Once the new nightly version appears in `firefox/nightly/latest-mozilla-central/ <https://archive.mozilla.org/pub/firefox/nightly/latest-mozilla-central/>`__, we can update the nightly version and release dates in ShipIt.
-**Bumping the Firefox nightly version and release dates in ShipIt should be done on release day.**
-This way, we can be confident that the first nightly builds with the new version will be available to ShipIt.
-
-**WARNING: Do not bump the nightly version or release dates in ShipIt until the new nightly builds are available.**
+In ShipIt, the Firefox nightly version and the release dates are hard-coded, and a human must update them when new nightly builds become available. These steps can be done on release day (aka, the day after the merge was performed). By that time, there should be at least one new Nightly version available.
```
[the remark this answers is not in the document]
Comment (bhearsum) [reply procedures/release-duty/merge-duty/merge_duty.rst:388]: Ah, okay. Sounds good.

[the remark this answers is not in the document]
Comment (bhearsum) [reply procedures/release-duty/merge-duty/merge_duty.rst:387]: Added a note for the next cycle about this - what do you think?
